### PR TITLE
Login logout 14

### PR DESCRIPTION
**Pull request by muydanny**
Cant login with invalid credentials

User story complete

Already logged in redirected-regular user

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -1,20 +1,26 @@
 class SessionsController < ApplicationController
 
   def new  
+    if current_user
+      route = determine_route(current_user)
```
Comment (stellakunzang) [inline app/controllers/sessions_controller.rb:5]: sexy helper method, whassup?!

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -1,20 +1,26 @@
...
       session[:user_id] = user.id
       flash[:success] = "Welcome, #{user.name}! You are now logged in."
 
       route = determine_route(user)
 
       redirect_to route
     else
-      flash[:error] = "Sorry, youre not one of us."
+      redirect_to "/login"
+      flash[:error] = "Sorry, your credentials were incorrect."
```
Comment (sagemlee) [inline app/controllers/sessions_controller.rb:23]: I am reading this error message in your voice

File: app/controllers/welcome_controller.rb [hunk 3]
```diff
@@ -1,5 +1,18 @@
...
+      redirect_to route
+    end
+  end
 
+  def determine_route(user)
+    {
+      "admin" => "/admin",
+      "merchant" => "/merchant",
+      "default" => "/profile"
+    }[user.role]
```
Comment (muydanny) [inline app/controllers/welcome_controller.rb:16]: My Mona Lisa

File: app/controllers/welcome_controller.rb [hunk 4]
```diff
@@ -1,5 +1,18 @@
...
   end
```
Comment (Gallup93) [inline app/controllers/welcome_controller.rb:17]: If i knew how to properly comment a gif, I'd be giving you a sweet gif right here. Good work

File: app/controllers/welcome_controller.rb [hunk 5]
```diff
@@ -1,5 +1,18 @@
...
+      route = determine_route(current_user)
+
+      redirect_to route
+    end
+  end
 
+  def determine_route(user)
+    {
+      "admin" => "/admin",
+      "merchant" => "/merchant",
```
Comment (sagemlee) [inline app/controllers/welcome_controller.rb:14]: awesome helper method.

**Review by stellakunzang — APPROVED** [review]

File: spec/features/sessions/regular_user_new_spec.rb [hunk 6]
```diff
@@ -27,8 +27,62 @@
...
+
+# As a registered user, merchant, or admin
+# When I visit the login path
+# If I am a regular user, I am redirected to my profile page
+# If I am a merchant user, I am redirected to my merchant dashboard page
+# If I am an admin user, I am redirected to my admin dashboard page
+# And I see a flash message that tells me I am already logged in
+
+# User Story 14
+
```
Comment (sagemlee) [inline spec/features/sessions/regular_user_new_spec.rb:79]: We should remember to remove this when we refactor.

**Review by sagemlee — APPROVED** [review]

File: app/controllers/welcome_controller.rb (hunk 3 above)
Comment (stellakunzang) [reply app/controllers/welcome_controller.rb:16]: I'm proud of you

File: app/controllers/welcome_controller.rb (hunk 4 above)
Comment (Gallup93) [reply app/controllers/welcome_controller.rb:17]: Thank you Stella!!!

**Pull request merged**